test/eredis_parser_tests.erl

```diff
@@ -36,134 +34,134 @@ one_byte_parse(S, <<Byte, B/binary>>) ->
+    ?assertEqual({ok, <<"bar">>, #pstate{}}, eredis_parser:parse(#pstate{}, B)).
 
 parse_split_bulk_test() ->
-    State1 = init(),
+    State1 = eredis_parser:init(),
     B1 = <<"$3\r\n">>,
     B2 = <<"bar\r\n">>,
 
-    {continue, State2} = parse(State1, B1),
-    Buffer = buffer_create(<<"\r\n">>),
+    {continue, State2} = eredis_parser:parse(State1, B1),
+    Buffer = eredis_parser:buffer_create(<<"\r\n">>),
     ?assertEqual(#pstate{state = bulk_continue, continuation_data = {3, Buffer}},
                  State2),
 
-    ?assertMatch({ok, <<"bar">>, _}, parse(State2, B2)).
+    ?assertMatch({ok, <<"bar">>, _}, eredis_parser:parse(State2, B2)).
 
 
 parse_very_split_bulk_test() ->
-    State1 = init(),
+    State1 = eredis_parser:init(),
     B1 = <<"$1">>,
     B2 = <<"3\r\n">>,
     B3 = <<"foobarbazquux\r\n">>, %% 13 bytes
 
-    Buffer1 = buffer_create(<<"$1">>),
+    Buffer1 = eredis_parser:buffer_create(<<"$1">>),
     ?assertEqual({continue,
                   #pstate{state = bulk_continue,
                           continuation_data = {incomplete_size, Buffer1}}},
-                 parse(State1, B1)),
-    {continue, State2} = parse(State1, B1),
+                 eredis_parser:parse(State1, B1)),
+    {continue, State2} = eredis_parser:parse(State1, B1),
 
-    Buffer2 = buffer_create(<<"\r\n">>),
+    Buffer2 = eredis_parser:buffer_create(<<"\r\n">>),
     ?assertEqual({continue,
                   #pstate{state = bulk_continue,
                           continuation_data = {13, Buffer2}}},
-                 parse(State2, B2)),
-    {continue, State3} = parse(State2, B2),
+                 eredis_parser:parse(State2, B2)),
+    {continue, State3} = eredis_parser:parse(State2, B2),
 
-    ?assertMatch({ok, <<"foobarbazquux">>, _}, parse(State3, B3)).
+    ?assertMatch({ok, <<"foobarbazquux">>, _}, eredis_parser:parse(State3, B3)).
 
 
 too_much_data_test() ->
     B = <<"$1\r\n1\r\n$1\r\n2\r\n$1\r\n3\r\n">>,
-    ?assertEqual({ok, <<"1">>, <<"$1\r\n2\r\n$1\r\n3\r\n">>}, parse_bulk(B)).
+    ?assertEqual({ok, <<"1">>, <<"$1\r\n2\r\n$1\r\n3\r\n">>}, eredis_parser:parse_bulk(B)).
 
 too_much_data_in_continuation_test() ->
     B1 = <<"$1\r\n">>,
     B2 = <<"1\r\n$1\r\n2\r\n$1\r\n3\r\n">>,
 
-    Buffer = buffer_create(<<"\r\n">>),
-    ?assertEqual({continue, {1, Buffer}}, parse_bulk(B1)),
-    {continue, ContinuationData1} = parse_bulk(B1),
+    Buffer = eredis_parser:buffer_create(<<"\r\n">>),
+    ?assertEqual({continue, {1, Buffer}}, eredis_parser:parse_bulk(B1)),
+    {continue, ContinuationData1} = eredis_parser:parse_bulk(B1),
 
     ?assertEqual({ok, <<"1">>, <<"$1\r\n2\r\n$1\r\n3\r\n">>},
-                 parse_bulk(ContinuationData1, B2)).
+                 eredis_parser:parse_bulk(ContinuationData1, B2)).
 
 bulk_test_() ->
     B = <<"$3\r\nbar\r\n">>,
-    ?_assertEqual({ok, <<"bar">>, <<>>}, parse_bulk(B)).
+    ?_assertEqual({ok, <<"bar">>, <<>>}, eredis_parser:parse_bulk(B)).
```

Comment:
^ What's `?_assertEqual`, where does it come from, and how does it differ from `?assertEqual`?

(Same for whatever occurrences there are of it, lest I repeat myself up to infinity and back again.)

Reply:
I don't know where the macro comes from. If you look closely it's outside my scope of change 😄

Reply:
(but, worry not, I'll fetch that for you)

Reply:
From OTP's `eunit.hrl`

```erlang
-define(_assertEqual(Expect, Expr), ?_test(?assertEqual(Expect, Expr))).
```

Reply:
What is `_test` you ask?

I'm glad you did
```erlang
-define(_test(Expr), {?LINE, fun () -> (Expr) end}).
```

Reply:
I guess `?LINE` you know about 😉